compiler/record-hessian/include/record-hessian/HessianComputer.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __RECORD_HESSIAN_HESSIANCOMPUTER_H__
+#define __RECORD_HESSIAN_HESSIANCOMPUTER_H__
+
+#include "record-hessian/HessianVector.h"
+
+#include <luci/IR/CircleNode.h>
+#include <luci_interpreter/Interpreter.h>
+
+namespace record_hessian
+{
+/**
+ * @brief Record approximated hessian matrix from GPTQ paper(https://arxiv.org/abs/2210.17323).
+ */
+using HessianMap = std::unordered_map<const luci::CircleNode *, std::vector<float>>;
+
+class HessianComputer
+{
+public:
+  // Record min/max of node
+  void recordHessian(const luci::CircleNode *node, const luci_interpreter::Tensor *input_tensor);
+
+  std::unique_ptr<HessianMap> getMap();
+
+private:
+  std::unordered_map<const luci::CircleNode *, HessianVector> _hessian_map;
```

Comment:
It's difficult to distinguish.
1/ it would be also better add alias for `std::unordered_map<const luci::CircleNode *, HessianVector>`
2/ but the name looks good with `HessianMap`
3/ then need new name for `std::unordered_map<const luci::CircleNode *, std::vector>`